src/app/context.tsx

```diff
@@ -0,0 +1,27 @@
+'use client';
+
+import React, { ReactNode, createContext, useContext, useState } from 'react';
+import { NewsWithId } from './models/types';
+
+type NewsState = {
+  context: NewsWithId[] | null;
+  setContext(news: NewsWithId[]): void;
+};
+
+const initialNewsState = {
```

Comment:
Why do you need this variable? You could put this object directly into `createContext`, couldn’t you?